apps/worker/src/batch_announcer/ipfs.announcer.ts

```diff
@@ -1,16 +1,334 @@
+      published: content.profile.published || '',
+      location: {
+        latitude: content.profile.location?.latitude,
+        longitude: content.profile.location?.longitude,
+        radius: content.profile.location?.radius,
+        altitude: content.profile.location?.altitude,
+        accuracy: content.profile.location?.accuracy,
+        name: content.profile.location?.name || '',
+        type: 'Place',
+      },
+      summary: content.profile.summary || '',
+      icon: attachments,
+    };
+    const profileString = JSON.stringify(profileActivity);
+    const [cid, hash] = await this.pinStringToIPFS(Buffer.from(profileString));
+    const profileAnnouncement = createProfile(dsnpUserId, await this.formIpfsUrl(cid), hash);
+    await writer.appendRow(profileAnnouncement);
+  }
+
+  private async bufferPublishStream(publishStream: PassThrough): Promise<Buffer> {
+    this.logger.debug('Buffering publish stream');
+    return new Promise((resolve, reject) => {
+      const buffers: Buffer[] = [];
+      publishStream.on('data', (data) => {
+        buffers.push(data);
+      });
+      publishStream.on('end', () => {
+        resolve(Buffer.concat(buffers));
+      });
+      publishStream.on('error', (err) => {
+        reject(err);
+      });
+    });
+  }
+
+  public async prepareNote(noteContent?: any): Promise<[string, string, string]> {
+    this.logger.debug(`Preparing note`);
+    const tags: ActivityContentTag[] = [];
+    if (noteContent?.content.tag) {
+      noteContent.content.tag.forEach((tag) => {
+        switch (tag.type) {
+          case TagTypeDto.Hashtag:
+            tags.push({ name: tag.name || '' });
+            break;
+          case TagTypeDto.Mention:
+            tags.push({
+              name: tag.name || '',
+              type: 'Mention',
+              id: tag.mentionedId || '',
+            });
+            break;
+          default:
+            throw new Error(`Unsupported tag type ${typeof tag.type}`);
+        }
+      });
+    }
+
+    const attachments: ActivityContentAttachment[] = [];
+    if (noteContent?.content.assets) {
+      noteContent.content.assets.forEach(async (asset: AssetDto) => {
+        switch (asset.type) {
+          case AttachmentTypeDto.LINK: {
+            const link: ActivityContentLink = {
+              type: 'Link',
+              href: asset.href || '',
+              name: asset.name || '',
+            };
+
+            attachments.push(link);
+            break;
+          }
+          case AttachmentTypeDto.IMAGE: {
+            const imageLinks: ActivityContentImageLink[] = [];
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              const image: ActivityContentImageLink = {
+                mediaType: 'image', // TODO
+                hash: [hashedContent],
```

Comment:
@aramikm  can we also send mediaType in reference dto?

Reply:
we are going to create these in request processor which we already have access to mediaType